resource-service/common/go-git.go

```diff
@@ -17,40 +26,55 @@ func (t GogitReal) PlainOpen(path string) (*git.Repository, error) {
 	return git.PlainOpen(path)
 }
 
-func (t GogitReal) PlainClone(path string, isBare bool, o *git.CloneOptions) (*git.Repository, error) {
-	return git.PlainClone(path, isBare, o)
+func (t GogitReal) PlainClone(gitContext common_models.GitContext, path string, isBare bool, o *git.CloneOptions) (*git.Repository, error) {
+	repo, err := git.PlainClone(path, isBare, o)
+	if err != nil {
+		log.Warnf("Could not clone using go-git library: %v", err)
+		if strings.Contains(err.Error(), unexpectedClientErrorMessage) {
+			git2 := Git2Go{}
+			log.Debug("Try to clone using libgit2go")
+			return git2.PlainClone(gitContext, path, isBare, o)
+		}
+		return nil, err
+	}
+	return repo, nil
 }
 
 func (t GogitReal) PlainInit(path string, isBare bool) (*git.Repository, error) {
 	return git.PlainInit(path, isBare)
 }
 
-/*//go:generate moq -pkg common_mock -skip-ensure -out ./fake/gogitrepo_mock.go . Repository
-type Repository interface {
-	Remote(name string) (*git.Remote, error)
-	Remotes() ([]*git.Remote, error)
-	CreateRemote(c *config.RemoteConfig) (*git.Remote, error)
-	CreateRemoteAnonymous(c *config.RemoteConfig) (*git.Remote, error)
-	DeleteRemote(name string) error
-	Branch(name string) (*config.Branch, error)
-	CreateBranch(c *config.Branch) error
-	DeleteBranch(name string) error
-	resolveToCommitHash(h plumbing.Hash) (plumbing.Hash, error)
-	clone(ctx context.Context, o *git.CloneOptions) error
-	calculateRemoteHeadReference(spec []config.RefSpec, resolvedHead *plumbing.Reference) []*plumbing.Reference
-	Push(o *git.PushOptions) error
-	Branches() (storer.ReferenceIter, error)
-	TreeObject(h plumbing.Hash) (*object.Tree, error)
-	TreeObjects() (*object.TreeIter, error)
-	CommitObject(h plumbing.Hash) (*object.Commit, error)
-	CommitObjects() (object.CommitIter, error)
-	BlobObject(h plumbing.Hash) (*object.Blob, error)
-	BlobObjects() (*object.BlobIter, error)
-	Object(t plumbing.ObjectType, h plumbing.Hash) (object.Object, error)
-	Objects() (*object.ObjectIter, error)
-	Head() (*plumbing.Reference, error)
-	Reference(name plumbing.ReferenceName, resolved bool) (*plumbing.Reference, error)
-	References() (storer.ReferenceIter, error)
-	Worktree() (*git.Worktree, error)
-	ResolveRevision(rev plumbing.Revision) (*plumbing.Hash, error)
-}*/
+type Git2Go struct{}
+
+func (g Git2Go) PlainClone(gitContext common_models.GitContext, path string, isBare bool, o *git.CloneOptions) (*git.Repository, error) {
+	_, err := git2go.Clone(o.URL, path, &git2go.CloneOptions{
```

Comment:
what variable we are ignoring and for which purpose?

Reply:
added some comments